src/request/xhr.ts

```diff
@@ -67,8 +69,15 @@ export default function xhr<T>(url: string, options: XhrRequestOptions = {}): Re
 
 				response.statusCode = request.status;
 				response.statusText = request.statusText;
-
-				resolve(response);
+				if (response.statusCode >= 200 && response.statusCode < 400) {
```

Comment:
Would this be better as just `response.statusCode < 400` since the 100 codes aren't errors?